src/Microsoft.OData.Client/Serialization/HttpWebRequestMessage.cs

```diff
@@ -63,149 +68,144 @@ public class HttpWebRequestMessage : DataServiceClientRequestMessage
+        /// <summary>
+        ///  Gets or set the credentials for this request.
+        /// </summary>
+        public override ICredentials Credentials
         {
             get
             {
-                List<KeyValuePair<string, string>> headerValues = new List<KeyValuePair<string, string>>(this.httpRequest.Headers.Count);
-                foreach (string headerName in this.httpRequest.Headers.AllKeys)
-                {
-                    string headerValue = this.httpRequest.Headers[headerName];
-                    headerValues.Add(new KeyValuePair<string, string>(headerName, headerValue));
-                }
-
-                return headerValues;
+                return _handler.Credentials;
             }
-        }
-
-        /// <summary> Returns the underlying HttpWebRequest </summary>
-        public System.Net.HttpWebRequest HttpWebRequest
-        {
-            get
+            set
             {
-                return this.httpRequest;
+                _handler.Credentials = value;
             }
         }
 
-        /// <summary>
-        /// Gets or set the credentials for this request.
-        /// </summary>
-        public override System.Net.ICredentials Credentials
-        {
-            get { return this.httpRequest.Credentials; }
-            set { this.httpRequest.Credentials = value; }
-        }
+
+#if !(NETCOREAPP1_0 || NETCOREAPP2_0)
 
         /// <summary>
         /// Gets or sets the timeout (in seconds) for this request.
         /// </summary>
         public override int Timeout
         {
-            get { return this.httpRequest.Timeout; }
-            set { this.httpRequest.Timeout = (int)Math.Min(Int32.MaxValue, new TimeSpan(0, 0, value).TotalMilliseconds); }
+            get
+            {
+                return (int)_client.Timeout.TotalSeconds;
+            }
+            set
+            {
+                _client.Timeout = new TimeSpan(0, 0, value);
+            }
         }
 
         /// <summary>
-        /// Gets or sets the read and write timeout (in seconds) for this request.
+        /// Gets or sets the read write timeout (in seconds) for this request.
+        /// Setting the value of the read write timeout to be similar to that of timeout value. 
+        /// This property will not be used as it is not available in HttpClient and getting rid of it 
+        /// Will break people who have been using it before. 
         /// </summary>
+        [Obsolete("This method is not found in HttpClient")]
         public override int ReadWriteTimeout
         {
-            get { return this.httpRequest.ReadWriteTimeout; }
-            set { this.httpRequest.ReadWriteTimeout = (int)Math.Min(Int32.MaxValue, new TimeSpan(0, 0, value).TotalMilliseconds); }
+            get
+            {
+                return (int)_client.Timeout.TotalSeconds;
+            }
+            set
+            {
+                _client.Timeout = new TimeSpan(0, 0, value);

```

Comment:
```suggestion
                Timeout = value;
```
To avoid code duplication